File: cms/server/pkg/user/workspace.go
package user

type Workspace struct {
	id      WorkspaceID
	name    string
	members *Members
}

func (t *Workspace) ID() WorkspaceID {
	return t.id
}

func (t *Workspace) Name() string {
	return t.name
}

func (t *Workspace) Members() *Members {
	return t.members
}

func (t *Workspace) IsPersonal() bool {
	return t.members.Fixed()
}

func (t *Workspace) Rename(name string) {
	t.name = name
}

File: cms/web/src/gql/fragments/workspace.ts
import { gql } from "@apollo/client";

import { integrationFragment } from "@reearth-cms/gql/fragments";

export const workspaceFragment = gql`
  fragment WorkspaceFragment on Workspace {
    id
    name
    members {
      ... on WorkspaceUserMember {
        user {
          id
          name
          email
        }
        userId
        role
      }
      ... on WorkspaceIntegrationMember {
        integration {
          ...integrationFragment
        }
        integrationRole: role
        active
        invitedBy {
          id
          name
          email
        }
        invitedById
      }
    }
    personal
  }
  ${integrationFragment}
`;

export default workspaceFragment;
